### ISSUE TITLE
website: decide what to name the arm release

### ISSUE THREAD

**Issue by bradfitz** [golang/go#13669]
@minux, please tell us what to name the GOARM=6 filename.

Can you summarize all the identifiers for ARM and tell us what they mean?

/cc @broady @davecheney

**Comment by minux** [comment]
ARM used to label both its CPU implementations and architectures
with numbers.
See https://en.wikipedia.org/wiki/List_of_ARM_microarchitectures.

The architecture has a uniform naming scheme: ARMvN with some
suffixes. A larger architecture version number means more features.
For example, we support ARMv5E and above. And ARMv8-A is the
64-bit ISA.

The implementation of an architecture used to use the name matching
ARM\d+[A-Z-]*, and the number has no direct correspondence to
the architecture it implements. For example, ARM920 implements
ARMv4T but ARM926E implements ARMv5E.

To make matters worse, people frequently abbreviate the core name
to a single digit, e.g. ARM7/ARM9. This is where the ambiguity comes
from.

ARM has since switched to a new naming scheme for cores, the
Cortex-[ARM]\d+ scheme. But still, you can't infer the architectural
version purely from the number alone. (Cortex-A17 is ARMv7A,
Cortex-A35 is ARMv8-A, Cortex-M0 is ARMv6M, Cortex-M3 is v7M).

Similarly, the FPU in ARM cores also have architectural version.
VFPvN. But the problem that the FPU and CPU version are fully
independent. Fortunately, most application ARM cores contain
at least VFPv1 now.


For Go, GOARM roughly corresponds to ARM architectural version.
GOARM=5 means ARMv5E (or higher) without FPU,
GOARM=6 means ARMv6 (or higher) with FPU (VFPv1 or higher),
GOARM=7 means ARMv7 (or higher) with FPU (VFPv3 or higher).
The difference between GOARM=6 and GOARM=7 is really
small (actually, only a single new VFP instruction is used for 7,
and runtime uses some new barrier instructions for OSes other
than linux where kernel helper is not available.)


I suggest that we name the Go 1.6 linux/arm binary release as
follows:
GOARM=6: go1.6.linux-armv6l.tar.gz
(Or go1.6-linux-arm-v6l.tar.gz depending on whether we want to
separate the GOARCH value from the suffix or not.)
armv6l also matches the `uname -m` output on Linux.

The 'l' here stands for little endian. (I know at least one case where
a ARMv6 core is used in big endian mode.)

**Comment by rsc** [comment]
Sounds like this is done. Reopen if not.
